src/nvim/regexp_nfa.c

```diff
@@ -5969,7 +5964,7 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
   nfa_match = false;
 
   // Allocate memory for the lists of nodes.
-  size_t size = (prog->nstate + 1) * sizeof(nfa_thread_T);
+  size_t size = (size_t)(prog->nstate + 1) * sizeof(nfa_thread_T);
```

Comment:
Will PVS complain about this?

Reply:
Perhaps it will not as `size_t` is unsigned.

Reply:
I tested it locally, PVS will indeed create a warning for this. Nice catch.

Reply:
#19521